crates/solidity/inputs/language/src/dsl.rs

```diff
@@ -0,0 +1,915 @@
+    ) ;
+
+    scanner FixedTypeSize = ((('0' .. '9') +) 'x' (('0' .. '9') +)) ;
+
+    scanner ForKeyword = "for" ;
+
+    scanner FromKeyword = "from" ;
+
+    scanner FunctionKeyword = "function" ;
+
+    scanner GlobalKeyword = "global" ;
+
+    scanner GreaterThan = '>' ;
+
+    scanner GreaterThanEqual = ">=" ;
+
+    scanner GreaterThanGreaterThan = ">>" ;
+
+    scanner GreaterThanGreaterThanEqual = ">>=" ;
+
+    scanner GreaterThanGreaterThanGreaterThan = ">>>" ;
+
+    scanner GreaterThanGreaterThanGreaterThanEqual = ">>>=" ;
+
+    scanner GweiKeyword = { enabled from "0.6.11" "gwei" } ;
+
+    scanner HexByteEscape = ('x' (HexCharacter) (HexCharacter)) ;
+
+    scanner HexCharacter = (('0' .. '9') | ('A' .. 'F') | ('a' .. 'f')) ;
+
+    scanner HexLiteral = (("0x" | {disabled from "0.5.0" "0X"}) ((HexCharacter) +) (('_' ((HexCharacter) +)) *)) ;
+
+    scanner HexStringContents = ((HexCharacter) (HexCharacter) ((('_' ?) (HexCharacter) (HexCharacter)) *)) ;
+
+    scanner HexStringLiteral = (SingleQuotedHexStringLiteral | DoubleQuotedHexStringLiteral) ;
+
+    scanner HoursKeyword = "hours" ;
+
+    scanner Identifier = RawIdentifier ;
+
+    scanner IdentifierPart = (IdentifierStart | ('0' .. '9')) ;
+
+    scanner IdentifierStart = ('$' | ('A' .. 'Z') | '_' | ('a' .. 'z')) ;
+
+    scanner IfKeyword = "if" ;
+
+    scanner ImmutableKeyword = { enabled from "0.6.5" "immutable" } ;
+
+    scanner ImportKeyword = "import" ;
+
+    scanner IndexedKeyword = "indexed" ;
+
+    scanner IntegerTypeSize = (
+        "104" | "112" | "120" | "128" | "136" | "144" | "152" | "16" | "160" | "168" | "176" | "184" | "192" | "200" | "208" |
+        "216" | "224" | "232" | "24" | "240" | "248" | "256" | "32" | "40" | "48" | "56" | "64" | "72" | "8" | "80" | "88" | "96"
+    ) ;
+
+    scanner InterfaceKeyword = "interface" ;
+
+    scanner InternalKeyword = "internal" ;
+
+    scanner IsKeyword = "is" ;
+
+    // scanner KeywordInAnyVersion = (FixedBytesType | SignedFixedType | UnsignedFixedType | SignedIntegerType | UnsignedIntegerType | AddressKeyword | AnonymousKeyword | AsKeyword | AssemblyKeyword | BoolKeyword | BreakKeyword | CaseKeyword | ConstantKeyword | ContinueKeyword | ContractKeyword | DaysKeyword | DefaultKeyword | DeleteKeyword | DoKeyword | ElseKeyword | EnumKeyword | EtherKeyword | EventKeyword | ExternalKeyword | FalseKeyword | ForKeyword | FunctionKeyword | HoursKeyword | IfKeyword | ImportKeyword | IndexedKeyword | InterfaceKeyword | InternalKeyword | IsKeyword | LetKeyword | LibraryKeyword | MappingKeyword | MemoryKeyword | MinutesKeyword | ModifierKeyword | NewKeyword | PayableKeyword | PragmaKeyword | PrivateKeyword | PublicKeyword | PureKeyword | ReturnKeyword | ReturnsKeyword | SecondsKeyword | StorageKeyword | StringKeyword | StructKeyword | SwitchKeyword | TrueKeyword | UsingKeyword | ViewKeyword | WeeksKeyword | WeiKeyword | WhileKeyword) ;
```

Comment:
IIUC, the new scaner no longer needs these, and they are automatically calculated. Should we remove the commented scanners?

Reply:
I left them there because this is where the `Reserved` and `Deprecated` information comes from - I spent a fun day with Remix doing that.

Reply:
Aha!